src/sgame/components/MiningComponent.h

```diff
@@ -47,15 +46,16 @@ class MiningComponent: public MiningComponentBase {
 		 */
 		static float InterferenceMod(float distance);
 
-		/**
-		 * @return Whether the miner is currently mining (and interfering with other miners).
-		 */
-		bool Active() { return active; }
+		struct Efficiencies {
```

Comment:
Given that their usage is mutually exclusive, would it be better to expose two functions to fetch these values?

Reply:
We have these cases:
- Blueprint miner: calculate only predicted efficiency
- Miner is constructed: need to update only predicted efficiencies
- Miner finishes construction: need to update only current efficiencies
- Miner dies: need to update both predicted and current efficiencies

I could have flags for which ones InformNeighbors should update, but that seemed overly complex so I stuck with always updating both, which is done together for efficiency.

Reply:
Got it.